registry/client/transport/transport.go

```diff
@@ -0,0 +1,147 @@
+package transport
```

Comment:
@dmcgowan see https://github.com/docker/docker/pull/13265/files#diff-27f83ee13313e01950af50e4393198a8R1 for some modifications